### ISSUE TITLE
Simple .Net 8 project is not building for iOS with error : clang++ exited with code 1

### ISSUE THREAD

**Issue by miroatp** [AdamEssenmacher/GoogleApisForiOSComponents#30]
Hi Adam,

Thank you for your amazing work on these bindings! 

I wanted to give it a try in our Net 8.0 MAUI app but we're facing issues building it for iOS. It may be similar to other issues listed here as they often mention missing GAppM-10.24.0 dll, but I wasn't 100% sure hence creating a new issue.

I uploaded the simple app to a repo and added the terminal output and my env info to its Readme file. https://github.com/miroatp/maui-test/tree/main

In short the build fails with "error : clang++ exited with code 1" and mentions that it "Could not find or use auto-linked framework 'GoogleAppMeasurement'".

Any chance you could have a look at it or point me to some other .Net 8 MAUI demo app that works with these new Analytics/Messaging packages?

**Comment by bengavin** [comment]
I'd suggest making sure you've updated to the latest .NET MAUI build [8.0.60 as of today] and .NET 8 SDK [8.0.6/8.0.300], then ensure all the correct workloads are installed [dotnet workload install ios maui-ios] to see if that helps anything.  I'm assuming you also have installed the latest Xcode for your OS, but doesn't hurt to check that too.

**Comment by miroatp** [comment]
Thanks Ben for your reply, I double checked the tools and they are updated.

```
miroslavkouril@Admins-MBP maui-test % /usr/bin/xcodebuild -version
Xcode 15.4
Build version 15F31d
miroslavkouril@Admins-MBP maui-test % dotnet workload list        

Installed Workload Id      Manifest Version       Installation Source
---------------------------------------------------------------------
maui-ios                   8.0.40/8.0.100         SDK 8.0.300        
maui-android               8.0.40/8.0.100         SDK 8.0.300        
ios                        17.2.8053/8.0.100      SDK 8.0.300        
maui                       8.0.40/8.0.100         SDK 8.0.300        
android                    34.0.113/8.0.100       SDK 8.0.300        

Use `dotnet workload search` to find additional workloads to install.   
```

I tested the build on my mac via Visual Studio Code using the MAUI extension and also directly from the command line
```
dotnet restore ATPWTAJointApp/ATPWTAJointApp.csproj --force --force-evaluate --no-cache
dotnet build ATPWTAJointApp/ATPWTAJointApp.csproj --no-incremental --framework net8.0-ios --configuration Debug --property:Platform="iPhoneSimulator" --property:BuildIpa=false
```

The result is always the same error:
```
miroslavkouril@Admins-MBP maui-test % dotnet build ATPWTAJointApp/ATPWTAJointApp.csproj --no-incremental --framework net8.0-ios --configuration Debug --property:Platform="iPhoneSimulator" --property:BuildIpa=false
  Determining projects to restore...
  All projects are up-to-date for restore.
  Detected signing identity:
          
    Bundle Id: com.atptour.atptour-staging
    App Id: com.atptour.atptour-staging
  ATPWTAJointApp -> /Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/bin/iPhoneSimulator/Debug/net8.0-ios/iossimulator-arm64/ATPWTAJointApp.dll
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/tools/msbuild/iOS/Xamarin.Shared.targets(146,3): warning : The directory '/Users/miroslavkouril/Library/Caches/XamarinBuildDownload/GAppM-10.24.0/GoogleAppMeasurement-10.24.0/Frameworks' does not exist. [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
  Optimizing assemblies for size may change the behavior of the app. Be sure to test after publishing. See: https://aka.ms/dotnet-illink
  Optimizing assemblies for size. This process might take a while.
/Users/miroslavkouril/Library/Caches/XamarinBuildDownload/FAnlytcs-10.24.0/FirebaseAnalytics-10.24.0/Frameworks/FirebaseAnalytics.xcframework/ios-arm64_x86_64-simulator/FirebaseAnalytics.framework/FirebaseAnalytics : warning MT7091: The framework /Users/miroslavkouril/Library/Caches/XamarinBuildDownload/FAnlytcs-10.24.0/FirebaseAnalytics-10.24.0/Frameworks/FirebaseAnalytics.xcframework/ios-arm64_x86_64-simulator/FirebaseAnalytics.framework is a framework of static libraries, and will not be copied to the app. [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
  Tool xcrun execution finished (exit code = 1).
          
  ld: warning: Could not find or use auto-linked framework 'GoogleAppMeasurement'
  Undefined symbols for architecture arm64:
    "_APMAnalyticsConfiguration", referenced from:
        +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics.o)
    "_APMAppMeasurementOriginFirebase", referenced from:
        +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics.o)
    "_APMConsentSettings3P", referenced from:
        +[FIRAnalytics setConsent:] in FirebaseAnalytics(FIRAnalytics.o)
    "_APMFormattedEventName", referenced from:
        +[FIRAnalytics logEventWithOrigin:name:parameters:] in FirebaseAnalytics(FIRAnalytics.o)
    "_APMFormattedUserPropertyName", referenced from:
        +[FIRAnalytics setUserPropertyString:forName:] in FirebaseAnalytics(FIRAnalytics.o)
    "_APMIsAnalyticsCollectionDeactivated", referenced from:
        +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics.o)
    "_APMIsAnalyticsCollectionEnabled", referenced from:
        +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics.o)
    "_APMMonitorLogTagOptionKey", referenced from:
        +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics.o)
    "_APMUserDataFieldEmailAddress", referenced from:
        +[FIRAnalytics initiateOnDeviceConversionMeasurementWithEmailAddress:] in FirebaseAnalytics(FIRAnalytics.o)
    "_APMUserDataFieldHashedEmailAddress", referenced from:
        +[FIRAnalytics initiateOnDeviceConversionMeasurementWithHashedEmailAddress:] in FirebaseAnalytics(FIRAnalytics.o)
    "_APMUserDataFieldHashedPhoneNumber", referenced from:
        +[FIRAnalytics initiateOnDeviceConversionMeasurementWithHashedPhoneNumber:] in FirebaseAnalytics(FIRAnalytics.o)
    "_APMUserDataFieldPhoneNumber", referenced from:
        +[FIRAnalytics initiateOnDeviceConversionMeasurementWithPhoneNumber:] in FirebaseAnalytics(FIRAnalytics.o)
    "_OBJC_CLASS_$_APMAdExposureReporter", referenced from:
        _OBJC_CLASS_$_FIRAAdExposureReporter in FirebaseAnalytics(FIRAAdExposureReporter.o)
    "_OBJC_CLASS_$_APMAnalytics", referenced from:
        objc-class-ref in FirebaseAnalytics(FIRAnalytics.o)
    "_OBJC_CLASS_$_APMConditionalUserProperty", referenced from:
        _OBJC_CLASS_$_FIRAConditionalUserProperty in FirebaseAnalytics(FIRAConditionalUserProperty.o)
    "_OBJC_CLASS_$_APMConditionalUserPropertyController", referenced from:
        _OBJC_CLASS_$_FIRAConditionalUserPropertyController in FirebaseAnalytics(FIRAConditionalUserPropertyController.o)
    "_OBJC_CLASS_$_APMEvent", referenced from:
        _OBJC_CLASS_$_FIRAEvent in FirebaseAnalytics(FIRAEvent.o)
    "_OBJC_CLASS_$_APMIdentifiers", referenced from:
        _OBJC_CLASS_$_FIRAIdentifiers in FirebaseAnalytics(FIRAIdentifiers.o)
    "_OBJC_CLASS_$_APMIdentity", referenced from:
        objc-class-ref in FirebaseAnalytics(FIRAnalytics.o)
    "_OBJC_CLASS_$_APMMeasurement", referenced from:
        objc-class-ref in FirebaseAnalytics(FIRAnalytics.o)
        objc-class-ref in FirebaseAnalytics(FIRAMeasurement.o)
        _OBJC_CLASS_$_FIRAMeasurement in FirebaseAnalytics(FIRAMeasurement.o)
    "_OBJC_CLASS_$_APMScreenViewReporter", referenced from:
        objc-class-ref in FirebaseAnalytics(FIRAScreenViewReporter.o)
        _OBJC_CLASS_$_FIRAScreenViewReporter in FirebaseAnalytics(FIRAScreenViewReporter.o)
    "_OBJC_CLASS_$_APMUserAttribute", referenced from:
        _OBJC_CLASS_$_FIRAUserAttribute in FirebaseAnalytics(FIRAUserAttribute.o)
    "_OBJC_CLASS_$_APMValue", referenced from:
        _OBJC_CLASS_$_FIRAValue in FirebaseAnalytics(FIRAValue.o)
    "_OBJC_METACLASS_$_APMAdExposureReporter", referenced from:
        _OBJC_METACLASS_$_FIRAAdExposureReporter in FirebaseAnalytics(FIRAAdExposureReporter.o)
    "_OBJC_METACLASS_$_APMConditionalUserProperty", referenced from:
        _OBJC_METACLASS_$_FIRAConditionalUserProperty in FirebaseAnalytics(FIRAConditionalUserProperty.o)
    "_OBJC_METACLASS_$_APMConditionalUserPropertyController", referenced from:
        _OBJC_METACLASS_$_FIRAConditionalUserPropertyController in FirebaseAnalytics(FIRAConditionalUserPropertyController.o)
    "_OBJC_METACLASS_$_APMEvent", referenced from:
        _OBJC_METACLASS_$_FIRAEvent in FirebaseAnalytics(FIRAEvent.o)
    "_OBJC_METACLASS_$_APMIdentifiers", referenced from:
        _OBJC_METACLASS_$_FIRAIdentifiers in FirebaseAnalytics(FIRAIdentifiers.o)
    "_OBJC_METACLASS_$_APMMeasurement", referenced from:
        _OBJC_METACLASS_$_FIRAMeasurement in FirebaseAnalytics(FIRAMeasurement.o)
    "_OBJC_METACLASS_$_APMScreenViewReporter", referenced from:
        _OBJC_METACLASS_$_FIRAScreenViewReporter in FirebaseAnalytics(FIRAScreenViewReporter.o)
    "_OBJC_METACLASS_$_APMUserAttribute", referenced from:
        _OBJC_METACLASS_$_FIRAUserAttribute in FirebaseAnalytics(FIRAUserAttribute.o)
    "_OBJC_METACLASS_$_APMValue", referenced from:
        _OBJC_METACLASS_$_FIRAValue in FirebaseAnalytics(FIRAValue.o)
  ld: symbol(s) not found for architecture arm64
  clang: error: linker command failed with exit code 1 (use -v to see invocation)
  
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error : clang++ exited with code 1: [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error : ld: warning: Could not find or use auto-linked framework 'GoogleAppMeasurement' [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error : Undefined symbols for architecture arm64: [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :   "_APMAnalyticsConfiguration", referenced from: [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :       +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics.o) [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :   "_APMAppMeasurementOriginFirebase", referenced from: [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :       +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics.o) [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :   "_APMConsentSettings3P", referenced from: [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :       +[FIRAnalytics setConsent:] in FirebaseAnalytics(FIRAnalytics.o) [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :   "_APMFormattedEventName", referenced from: [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :       +[FIRAnalytics logEventWithOrigin:name:parameters:] in FirebaseAnalytics(FIRAnalytics.o) [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :   "_APMFormattedUserPropertyName", referenced from: [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :       +[FIRAnalytics setUserPropertyString:forName:] in FirebaseAnalytics(FIRAnalytics.o) [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :   "_APMIsAnalyticsCollectionDeactivated", referenced from: [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :       +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics.o) [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :   "_APMIsAnalyticsCollectionEnabled", referenced from: [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :       +[FIRAna [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]

Build FAILED.

/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/tools/msbuild/iOS/Xamarin.Shared.targets(146,3): warning : The directory '/Users/miroslavkouril/Library/Caches/XamarinBuildDownload/GAppM-10.24.0/GoogleAppMeasurement-10.24.0/Frameworks' does not exist. [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/Users/miroslavkouril/Library/Caches/XamarinBuildDownload/FAnlytcs-10.24.0/FirebaseAnalytics-10.24.0/Frameworks/FirebaseAnalytics.xcframework/ios-arm64_x86_64-simulator/FirebaseAnalytics.framework/FirebaseAnalytics : warning MT7091: The framework /Users/miroslavkouril/Library/Caches/XamarinBuildDownload/FAnlytcs-10.24.0/FirebaseAnalytics-10.24.0/Frameworks/FirebaseAnalytics.xcframework/ios-arm64_x86_64-simulator/FirebaseAnalytics.framework is a framework of static libraries, and will not be copied to the app. [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error : clang++ exited with code 1: [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error : ld: warning: Could not find or use auto-linked framework 'GoogleAppMeasurement' [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error : Undefined symbols for architecture arm64: [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :   "_APMAnalyticsConfiguration", referenced from: [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :       +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics.o) [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :   "_APMAppMeasurementOriginFirebase", referenced from: [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :       +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics.o) [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :   "_APMConsentSettings3P", referenced from: [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :       +[FIRAnalytics setConsent:] in FirebaseAnalytics(FIRAnalytics.o) [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :   "_APMFormattedEventName", referenced from: [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :       +[FIRAnalytics logEventWithOrigin:name:parameters:] in FirebaseAnalytics(FIRAnalytics.o) [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :   "_APMFormattedUserPropertyName", referenced from: [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :       +[FIRAnalytics setUserPropertyString:forName:] in FirebaseAnalytics(FIRAnalytics.o) [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :   "_APMIsAnalyticsCollectionDeactivated", referenced from: [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :       +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics.o) [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :   "_APMIsAnalyticsCollectionEnabled", referenced from: [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.2.8053/targets/Xamarin.Shared.Sdk.targets(1560,3): error :       +[FIRAna [/Users/miroslavkouril/Documents/Development/maui-test/ATPWTAJointApp/ATPWTAJointApp.csproj::TargetFramework=net8.0-ios]
    2 Warning(s)
    1 Error(s)

Time Elapsed 00:00:24.47
```

**Comment by MateuszKapusta** [comment]
Probably problem with to long path. 
"In AdamE.Firebase.iOS.Core, GoogleAppMeasurement is unique in that it is the only dependency not packed directly into the nuget package. Instead, it is downloaded as part of the build by Xamarin.Build.Download."

After a few tries with an empty project GAppM-10.24 was download correctly.
You can copy this to "C:\Users\admin\AppData\Local\XamarinBuildDownloadCache"
[GAppM-10.24.0.zip](https://github.com/user-attachments/files/15911425/GAppM-10.24.0.zip)

**Comment by miroatp** [comment]
Thank you for testing it, I'll try to reduce the path length to see if it downloads the dependency automatically. If not I'll use the shared library. Either way it goes I'll put my update here afterwards.

**Comment by miroatp** [comment]
I can confirm that reducing the path length & deleting the cached framework ([User]/Library/Caches/XamarinBuildDownload/FAnlytcs-10.24.0) fixed the compile error and the app example builds fine now. Thanks all for your suggestions!

**Comment by datvm** [comment]
@miroatp How did you shorten the path length? I can't even paste the attached file into my Cache folder (`C:\Users\lukev\AppData\Local\XamarinBuildDownloadCache\GAppM-10.24.0`) because Windows says it's too long.

**Comment by MateuszKapusta** [comment]
@datvm You should be able to copy the zip file to your cache and then unzip it if this doesn't work you can always delete GAppM-10.24.0 and then try to rebuild the app, after a few attempts. It worked for me

**Comment by datvm** [comment]
@MateuszKapusta Thanks, I have already solved it. I copied it using Powershell instead of Windows Explorer:

```
Copy-Item -Path "C:\Users\<username>\Downloads\GAppM-10.24.0" -Destination "C:\Users\<username>\AppData\Local\XamarinBuildDownloadCache" -Recurse
```

**Comment by miroatp** [comment]
@datvm Sorry I was away on holidays :) The cache path wasn't the problem in my case, it was the source code path, so I just moved the app to something like ../Users/Miro/P/DemoApp.sln and opened it in VS to get the sdk downloaded...

**Comment by datvm** [comment]
@miroatp ah I see. Different issue then. Btw in that case in the future, you can simply build your project from the CLI (outside of VS) if you don't want to move it. Another solution is to use `BaseOutputPath` like [this](https://github.com/dotnet/maui/issues/10068#issuecomment-1826435176)

**Comment by yvbeek** [comment]
As this fix requires a couple of manual steps, you can also have msbuild handle it:
```props
<Target
  Name="FixMissingGAM"
  BeforeTargets="PrepareForBuild"
  Condition="!Exists('$(UserProfile)/Library/Caches/XamarinBuildDownload/GAppM-11.4.0/GoogleAppMeasurement-11.4.0)')">
    <Message Text="Fixing missing GoogleAppMeasurement folder" Importance="High" />
    <Exec Command="curl -L -o /tmp/GoogleAppMeasurement-11.4.0.tar.gz https://dl.google.com/firebase/ios/analytics/16e615e7058d8274/GoogleAppMeasurement-11.4.0.tar.gz" />
    <Exec Command="tar -xzf /tmp/GoogleAppMeasurement-11.4.0.tar.gz -C ~/Library/Caches/XamarinBuildDownload/GAppM-11.4.0/" />
    <Delete Files="/tmp/GoogleAppMeasurement-11.4.0.tar.gz" />
</Target>
```

Note: this implementation only works for macOS, but with a few conditions you could tweak it so that it works on Windows too
